plugin.py

```diff
@@ -9,32 +10,77 @@
 from LSP.plugin.core.protocol import Response
 from LSP.plugin.core.settings import ClientConfig
 from LSP.plugin.core.settings import read_client_config
+from sublime_lib import ResourcePath
 
 
-package_path = os.path.dirname(__file__)
-server_path = os.path.join(package_path, 'vscode-eslint', 'out', 'eslintServer.js')
-vscode_eslint_path = os.path.join(package_path, 'vscode-eslint')
-node_modules_path = os.path.join(vscode_eslint_path, 'node_modules')
+PACKAGE_NAME = 'LSP-eslint'
+SETTINGS_FILENAME = 'LSP-eslint.sublime-settings'
 
 
-def plugin_loaded():
-    dependencies_insalled = os.path.isdir(node_modules_path)
-    print('LSP-eslint: Server {} installed.'.format('is' if dependencies_insalled else 'is not'))
+class ServerResources(object):
+    """Global object providing paths to server resources. Also handles installing/updating server in cache.
+
+    It needs to be initialized during (or after) plugin_loaded() for it to return relevant and valid paths.
+    """
+    initialized = False
+    server_binary_path = None
+
+    @classmethod
+    def initialize(cls):
+        if cls.initialized:
+            return
+
+        cls.initialized = True
+
+        cache_path = sublime.cache_path()
+        server_path = os.path.join(cache_path, PACKAGE_NAME, 'server')
+        cls.server_binary_path = os.path.join(server_path, 'out', 'eslintServer.js')
+        cls.copy_resources(server_path)
+
+    @classmethod
+    def copy_resources(cls, server_path):
+        local_server_path = 'Packages/{}/vscode-eslint/'.format(PACKAGE_NAME)
+
+        if os.path.isdir(server_path):
+            # Server already in cache. Check if version has changed and if so,
+            # delete existing copy in cache.
+            delete_cached_server = False
+            cache_package_json_path = os.path.join(server_path, 'package.json')
+            if os.path.isfile(cache_package_json_path):
+                # Read package.json in cache
+                with open(cache_package_json_path, 'r') as f:
+                    cached_version = json.load(f)['version']
+                # Read local package.json
+                local_package_json_res = ResourcePath(local_server_path, 'package.json')
+                local_version = json.loads(local_package_json_res.read_text())['version']
+                delete_cached_server = cached_version != local_version
+            else:
```

Comment:
It is a shame that sublime.load_resource and ResourcePath only expect a string like 'Cache/LSP-eslint/server/package.json' to work, else they will throw a FileNotFoundError. 

Is that the reason you use two different ways to read files?
- `with open(cache_package_json_path, 'r') as f:`
-  ResourcePath(local_server_path, 'package.json')

It would be great if they also accepted a full path to the file, but they don't.
Do you think that this code bellow is more readable?

```
    @classmethod
    def copy_resources(cls, server_path):
        local_server_path = 'Packages/{}/vscode-eslint/'.format(PACKAGE_NAME)
        cache_server_path = 'Cache/{}/server/'.format(PACKAGE_NAME)

        if os.path.isdir(server_path):
            # Server already in cache. Check if version has changed and if so,
            # delete existing copy in cache.
            try:
                cache_package_json = json.loads(
                    ResourcePath(cache_server_path, 'package.json').read_text())

                local_package_json = json.loads(
                    ResourcePath(local_server_path, 'package.json').read_text())

                if cache_package_json['version'] != local_package_json['version']:
                    shutil.rmtree(server_path)

            except FileNotFoundError:
                shutil.rmtree(server_path)

        if not os.path.isdir(server_path):
            ResourcePath(local_server_path).copytree(server_path, exist_ok=True)
```

Reply:
> Is that the reason you use two different ways to read files?

I don't use sublime API (or ResourcePath) for Cache as I'm assuming it's a normal directory on a normal file system so I can just use pure file APIs (I can't, of course, do that for files in zipped packages). But I wouldn't mind switching to your suggestion since it makes things a bit cleaner. But of course we still have to use file API `shutil.rmtree` to delete whole directory.

Reply:
But then maybe it gets more confusing with to different paths to cache.

Reply:
I can commit the change if you want?

Reply:
> But then maybe it gets more confusing with two different paths to cache director.

Agree

Reply:
It would be good if I could pass the `server_path` to `ResourcePath(server_path, 'package.json').read_text())`, instead of craeting a new variable `cache_server_path`. But If I do that, it would throw a `FileNotFoundError`.

I am ok with the way you wrote it. I just found this part a bit harder to read, so I wanted to find a way to make it a bit readable, but the cost is introducing a new confusing variable...

So I have nothing against getting this merged :)